lib/irclib/client.js

```diff
@@ -85,13 +94,22 @@ BridgedClient.prototype.connect = function(callbacks) {
             "Connecting to IRC server %s as %s (user=%s)",
             server.domain, nameInfo.nick, nameInfo.username
         );
+        userCount++;
 
-        return clientConnection.create(server, {
+        var opts = {
             nick: nameInfo.nick,
             username: nameInfo.username,
             realname: nameInfo.realname,
             password: self.password
-        }, function(inst) {
+        };
+        if (ip6Prefix) {
+            var suffix = userCount.toString(16);
+            // insert colons: "deadbeef" => "dead:beef"
+            suffix = suffix.replace(/\B(?=(.{4})+(?!.))/g, ':');
```

Comment:
That doesn't work properly though --- it produces 'abcd:ef' when it should be 'ab:cdef'